Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py

```diff
@@ -1203,65 +1211,36 @@ def insert_to_updated_context(context_data: dict,
     return new_context_data, version
 
 
-def safely_update_context_data(
-    context_data: dict,
-    version: Any,
-    offense_ids: list | None = None,
-    should_update_last_fetch: bool = False,
-    should_update_last_mirror: bool = False,
-    should_add_reset_key: bool = False,
-    should_force_update: bool = False,
-) -> None:
-    """Safely updates context
-
-    Args:
-        context_data (dict): The context data to save (encoded)
-        version (Any): The context current version
-        offense_ids (list, optional): List of offenses ids to change. Defaults to None.
-        should_update_last_fetch (bool, optional): If we should update last fetch. Defaults to False
-        should_update_last_mirror (bool, optional): If we should update last mirror. Defaults to False
-        should_add_reset_key (bool, optional): If we should add reset key. Defaults to False
-        should_force_update (bool, optional): If we should force update the current context. Defaults to False
-
+def deep_merge_context_changes(current_ctx: dict, changes: dict) -> None:
+    """
+    Recursively merges 'changes' into 'current_ctx'.
+    If a value is a dict, merges sub-keys. Otherwise overwrites.
+    """
+    for k, v in changes.items():
+        if isinstance(v, dict) and isinstance(current_ctx.get(k), dict):
+            deep_merge_context_changes(current_ctx[k], v)
```

Comment:
let's talk about this.
I am not sure about the size and depths of the contex and if it coul be a performance issue.